pkg/test/client.go

```diff
@@ -182,6 +192,22 @@ func toMap(obj runtime.Object) (map[string]interface{}, error) {
 	return m, nil
 }
 
+func getAllToolchainResources(s *runtime.Scheme) []client.Object {
+	toolchainObjs := make([]client.Object, 0)
+	KindToTypeMap := s.KnownTypes(toolchainv1alpha1.GroupVersion)
+	var kinds []string
+	for key := range KindToTypeMap {
+		kinds = append(kinds, key)
+	}
+	for _, k := range kinds {
+		obj := &unstructured.Unstructured{}
+		gvk := schema.GroupVersionKind{Group: toolchainv1alpha1.GroupVersion.Group, Version: toolchainv1alpha1.GroupVersion.Version, Kind: k}
+		obj.SetGroupVersionKind(gvk)
+		toolchainObjs = append(toolchainObjs, obj)
+	}
+	return toolchainObjs
+}
```

Comment:
you can simplify it to:
```suggestion
func getAllToolchainResources(s *runtime.Scheme) []client.Object {
	kindToTypeMap := s.KnownTypes(toolchainv1alpha1.GroupVersion)
	toolchainObjs := make([]client.Object, 0, len(kindToTypeMap))
	for kind := range kindToTypeMap {
		obj := &unstructured.Unstructured{}
		obj.SetGroupVersionKind(toolchainv1alpha1.GroupVersion.WithKind(kind))
		toolchainObjs = append(toolchainObjs, obj)
	}
	return toolchainObjs
}
```